### ISSUE TITLE
G5 Module Instances don't work with Advanced Module Manager

### ISSUE THREAD

**Issue by topwebs** [gantry/gantry5#1685]
Basically, I want certain modules to only appear on the home page of the Mosets Tree component. So AMM has a condition to only show on home page, but with a G5 module instance, the module appears on every page of Mosets Tree. The URL condition also does not work to only show the module on the home page. Upon further investigation, it appears that Joomla module instances, also are not able to be shown on the home page only with AMM. Could G5 be changed in a way so that module instances would work with Advanced Module Manager? This is important for placing modules on component pages.

**Comment by mahagr** [comment]
I've just checked our code on AMM and it looks like we're calling modules in standard way and the hook that AMM is using is indeed getting called. Do any other rules from AMM work?

**Comment by mahagr** [comment]
BTW: AMM hooks into `onRenderModule` event; you can do the same from your own plugin:

```php
    /**
     * @param object $module
     * @param array $attribs
     */
    public function onRenderModule(&$module, &$attribs)
    {
        if (!$this->app->isSite()) {
            return;
        }

        // Set $module = null if you want to hide the module.
    }
```

**Comment by topwebs** [comment]
AMM Menu rules work, but not Home Page or Url match. 

My issue is with creating a module instance in G5. As far as I understand, this does not use the Joomla module position, as it doesn't matter what module position you use in the module manager. The position comes from G5. Perhaps this is the issue. If AMM uses the module position, then perhaps it does not know it from a G5 module instance. Does G5 communicate a module instance position to Joomla that AMM can pick up on?

**Comment by mahagr** [comment]
Oh, you are using *module instance*, not position?

Module instance works just like it would work when you are manually embedding a module by using Joomla's `{loadmodule mod_mymodule, Module Title}`. As it doesn't use module position but is manually embedded to the page, it doesn't really/fully obey assignments either.

Check what happens if you use `{loadmodule}` instead..?

**Comment by topwebs** [comment]
Yes, module instance. Not sure where you want me to try to use {loadmodule}. I think what you're saying is that AMM cannot affect the showing of a module instance, other than by menu links. 

It seems then that there is no way to show a module instance on a specific component page. 

The reason we want to use module instances, is to style sidebar modules in different ways. When you use a module position in G5, the entire sidebar is styled. That is another issue.

**Comment by mahagr** [comment]
In every place where Joomla has an editor, there's also a button [ Module +]. That allows you to inject module inside text, for example inside an article. I am guessing that AMM isn't able to fully affect those either..?

BTW: you can create module positions just like instances, though it may be an overkill.

**Comment by topwebs** [comment]
I didn't have time to check {loadmodule}, although I don't think AMM will work with that either.

However, you have solved my problem. I didn't realize that I could "create" a module position with the key in a module position particle in G5. Those "created" module positions are controlled with AMM.  So, I basically create a new module position with each module position particle and it essentially acts like a module instance particle. Thanks.

**Comment by topwebs** [comment]
Now I created a G5 custom HTML module with a {loadmodule} inside. Then I used AMM to create a rule for the loaded module. That worked. So, there is something different about how G5 handles module instances that doesn't work with AMM.

**Comment by mahagr** [comment]
Thanks for the update. I need to check the module instances at some point, but I believe that your issue has been fixed, though maybe not ideal way..?

**Comment by topwebs** [comment]
Yes, I can create new module positions that would mimic module instances. However, it would be good to check why G5 module instances don't work like {loadmodule} with AMM, as that is the most  logical way to work.
